### PR TITLE
Tb 3149 adapt settings widgets design

### PR DESCRIPTION
**Pull request by pro100svitlo**
### What 🕵️ 🔍

- This PR contains some changes to the xayn design, mainly related to the settings screen
- new colors, styles were added. <b>Partially</b>. Properly they will be added, renamed and sorted later, with the separate task.
- new button (ghost) was added (with docs)
- update settings related widgets
  - some components were temporarily deprecated. I left them in the codebase, cos later they will be added again.
  - changed style and colors
- update settings related example page

----------

### How to test 🥼 🔬

- compare design from figma

----------

### Screenshots 📸 📱

| Before | After  |
| ------ | ------ |
|  |  |
|  |  |
|  |  |

----------
### References 📝 🔗

- [new colors scheme](https://www.figma.com/file/V9uNU8Bw8pKkkO9tmKvH3o/Design-System-%7C-Xayn?node-id=2008%3A1129)
- [figma settings](https://www.figma.com/file/hGAst3K9ZQIHmQoCNjbgaZ/Discovery-App-%7C-Design?node-id=391%3A1139)
- [jira task](https://xainag.atlassian.net/jira/software/projects/TB/boards/52?selectedIssue=TB-3149)

----------

### CODE REVIEW

File: example/lib/screen/xayn_widgets/page/settings_page.dart [hunk 1]
```diff
@@ -58,19 +61,19 @@ class _PrivacySectionState extends State<_PrivacySection> {
...
-          _getBlockAds(),
-          _getCookies(),
+          _getAboutXayn(),
           _getEmptyCache(),
         ],
       );
 
+  // ignore: unused_element
   SettingsCardData _getBlockTrackers() {
     final tile = SettingsTileData(
```
Comment (joecks) [inline example/lib/screen/xayn_widgets/page/settings_page.dart:73]: ❓  whats about those ignored elements?

File: lib/src/linden/xcolors.dart [hunk 2]
```diff
@@ -160,4 +160,40 @@ class XColors {
       brightness == Brightness.dark ? dark : null;
 
   Brightness get brightness => _brightness;
+
+  /// right now we are in the middle of the design migration.
+  /// So far I added here some items, which I need for my tasks.
+  /// But later we will restructure all this file with proper items (names and structure)
+  //region NEW COLORS
```
Comment (joecks) [inline lib/src/linden/xcolors.dart:167]: 💅  You could extend this class i.e. `NewColors extends XColors` and then pass a boolean to `Linden(newColors:true)` that would switch between the implementations, so you don't need to duplicate code.

File: lib/src/widget/settings/data/settings_tile_action.dart [hunk 3]
```diff
@@ -64,6 +66,7 @@ class SettingsTileActionCircle extends SettingsTileAction {
       ];
 }
 
+@Deprecated('So far there is no useCases, but later it might come back')
```
Comment (joecks) [inline lib/src/widget/settings/data/settings_tile_action.dart:69]: 💅 Personally I would avoid colloquial language in such descriptions, i.e.: `Old Style Settings, might be removed`

File: lib/src/widget/settings/widget/settings_section.dart [hunk 4]
```diff
@@ -3,11 +3,12 @@ import 'package:xayn_design/xayn_design.dart';
 
 const _defaultAlignment = CrossAxisAlignment.start;
 
-/// Should be equal to [XSizes.unit4];
-const _defaultTopPadding = 32.0;
+/// Should be equal to [XSizes.unit3];
+const _defaultTopPadding = 24.0;
```
Comment (joecks) [inline lib/src/widget/settings/widget/settings_section.dart:7]: ❓  why can't we use XSizes?

**Review by joecks — APPROVED** [review]
Looks good. I just added some smaller suggestions.

File: example/lib/screen/xayn_widgets/page/settings_page.dart (hunk 1 above)
Comment (pro100svitlo) [reply example/lib/screen/xayn_widgets/page/settings_page.dart:73]: I decided to leave those elements in the codebase. 
From the talk with Adriel, I got that right now they are busy and do not have time for implementing all components, but soon they will renew them as well. So I decided to leave them and later just uncomment and modify them.

File: lib/src/linden/xcolors.dart (hunk 2 above)
Comment (pro100svitlo) [reply lib/src/linden/xcolors.dart:241]: It is not just about switching between old and new. 
Right now UI team do not have all the color mapping scheme between the old and new palette. So its more like they add something new on top.

Comment (pro100svitlo) [reply lib/src/linden/xcolors.dart:241]: But yes, I can try to do this as well 😉

File: lib/src/widget/settings/widget/settings_section.dart (hunk 4 above)
Comment (pro100svitlo) [reply lib/src/widget/settings/widget/settings_section.dart:7]: it is not `const`

File: lib/src/widget/settings/data/settings_tile_action.dart (hunk 3 above)
Comment (pro100svitlo) [reply lib/src/widget/settings/data/settings_tile_action.dart]: updated

**Pull request merged**